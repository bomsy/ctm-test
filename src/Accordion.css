@media all and (min-width: 320px) and (max-width : 1024px) {
  .nav-bar {
    display: none;
  }

  .cards ul {
    list-style-type: none;
    padding: 0;
  }

  .card header .card-title {
    font-weight: normal;
  }

  .card header {
    padding: 20px;
    background-image: -webkit-gradient(linear, 0% 0%, 0% 100%, from(#1a3c7f), to(#000034));
    border-top-right-radius: 10px;
    border-top-left-radius: 10px;
    font-size: 14px;
  }

  .card header i.arrow {
    display: none;
  }

  .card section .card-side-note span {
    font-size: 24px;
  }
  
  .card section .card-side-note div {
    font-size: 40px;
  }

  .card section {
    background: #ffffff;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
    padding: 40px;
    color: #000000;
    display: none;
  }

  .card section .card-content {
    vertical-align: top;
    margin-top: 10px;
    margin-left: 0;
    display: inline-block;
    width: 100%;
    font-size: 12px;
    text-align: left;
  }

  .card section .card-side-note {
    vertical-align: top;
    margin-left: 0;
    padding-left: 0;
    margin-top: 20px;
    display: inline-block;
    font-weight: bold;
    border-left: 0px solid #aaaaaa;
  }

  .cards li.card {
    padding-bottom: 10px;
  }
  
  .card header .card-apr {
    float: right;
  }
  
}

@media all and (min-width: 1024px) {
  .cards ul {
    list-style-type: none;
    padding: 0;
  }
  
  .cards li.card {
    padding-bottom: 20px;
  }
  
  .card section {
    background: #ffffff;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
    padding: 40px;
    color: #000000;
    display: none;
  }
  
  .card section .card-content {
    vertical-align: top;
    margin-left: 60px;
    display: inline-block;
    width: 40%;
  }
  
  .card section .card-side-note {
    vertical-align: top;
    margin-left: 40px;
    padding-left: 40px;
    display: inline-block;
    width: 20%;
    border-left: 1px solid #aaaaaa;
    font-weight: bold;
  }
  
  .card section .card-side-note span {
    font-size: 24px;
  }
  
  .card section .card-side-note div {
    font-size: 40px;
  }
  .card header {
    padding: 20px;
    background-image: -webkit-gradient(linear, 0% 0%, 0% 100%, from(#1a3c7f), to(#000034));
    border-top-right-radius: 10px;
    border-top-left-radius: 10px;
    font-size: 20px;
  }
  
  .card header .card-title {
    font-weight: bold;
  }
  
  .card header .card-apr {
    float: right;
  }
}

.card header.collapsed {
  border-bottom-right-radius: 10px;
  border-bottom-left-radius: 10px;
}
.card section.expanded {
  display: block;
}