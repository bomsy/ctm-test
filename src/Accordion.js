"use strict";
import React, { Component } from "react";

export default class Accordion extends Component {
  constructor() {
    super();
    this.state = {
      current: null
    };
    this.handleClick = this.handleClick.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ current: nextProps.cards[0] });
  }

  handleClick(card) {
    this.setState({ current: card });
  }

  isCurrent(card) {
    return this.state.current === card;
  }

  renderCard(card, index) {
    return (
      <li className="card" key={index} onClick={() => this.handleClick(card)}>
        <header className={!this.isCurrent(card) ? "collapsed" : ""}>
          <i className="arrow nav-icon" />
          <span className="card-title">
            {card.name}
          </span>
          <span className="card-apr">
            {card.apr}% APR
          </span>
        </header>
        <section className={this.isCurrent(card) ? "expanded" : ""}>
          <img src={"../assets/" + card.code + ".png"} />
          <article className="card-content">
            {card.information}
          </article>
          <aside className="card-side-note">
            <span>Cashback</span>
            <div>
              Â£{card.cashback}
            </div>
          </aside>
        </section>
      </li>
    );
  }

  render() {
    const { cards } = this.props;
    return (
      <div className="cards">
        <ul>
          {cards.map((card, index) => this.renderCard(card, index))}
        </ul>
      </div>
    );
  }
}
