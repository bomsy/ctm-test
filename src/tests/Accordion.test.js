import React from "react";
import Accordion from "../Accordion.js";
import renderer from "react-test-renderer";

test("Should load the accordion and collapse the first", () => {
  const cards = [
    {
      code: "CARD_274",
      name: "Barclaycard Platinum Credit Card",
      apr: 22.9,
      cashback: 120.0,
      information:
        "5% cashback during your first 3 months up to £2,000 in purchases (T&C's Apply)."
    },
    {
      code: "CARD_822",
      name: "Virgin Money Credit Card",
      apr: 28.2,
      cashback: 106.25,
      information:
        "5% cashback in first 3 months up to £125 (on the first £2,500 of purchases). After the first three months ongoing cashback is paid at 1.25%."
    }
  ];
  const component = renderer.create(<Accordion cards={cards} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
