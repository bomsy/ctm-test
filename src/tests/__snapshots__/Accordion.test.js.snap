// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should load the accordion and collapse the first 1`] = `
<div
  className="cards"
>
  <ul>
    <li
      className="card"
      onClick={[Function]}
    >
      <header
        className="collapsed"
      >
        <i
          className="arrow nav-icon"
        />
        <span
          className="card-title"
        >
          Barclaycard Platinum Credit Card
        </span>
        <span
          className="card-apr"
        >
          22.9
          % APR
        </span>
      </header>
      <section
        className=""
      >
        <img
          src="../assets/CARD_274.png"
        />
        <article
          className="card-content"
        >
          5% cashback during your first 3 months up to £2,000 in purchases (T&C's Apply).
        </article>
        <aside
          className="card-side-note"
        >
          <span>
            Cashback
          </span>
          <div>
            £
            120
          </div>
        </aside>
      </section>
    </li>
    <li
      className="card"
      onClick={[Function]}
    >
      <header
        className="collapsed"
      >
        <i
          className="arrow nav-icon"
        />
        <span
          className="card-title"
        >
          Virgin Money Credit Card
        </span>
        <span
          className="card-apr"
        >
          28.2
          % APR
        </span>
      </header>
      <section
        className=""
      >
        <img
          src="../assets/CARD_822.png"
        />
        <article
          className="card-content"
        >
          5% cashback in first 3 months up to £125 (on the first £2,500 of purchases). After the first three months ongoing cashback is paid at 1.25%.
        </article>
        <aside
          className="card-side-note"
        >
          <span>
            Cashback
          </span>
          <div>
            £
            106.25
          </div>
        </aside>
      </section>
    </li>
  </ul>
</div>
`;
